db/columns/operations/select.py

```diff
@@ -17,17 +17,24 @@
 DYNAMIC_NODE_TAGS = {"SQLValueFunction", "FuncCall"}
 
 
-def get_column_index_from_name(table_oid, column_name, engine, connection_to_use=None):
+def get_columns_attnum_from_names(table_oid, column_names, engine, connection_to_use=None):
     with warnings.catch_warnings():
         warnings.filterwarnings("ignore", message="Did not recognize type")
         pg_attribute = Table("pg_attribute", MetaData(), autoload_with=engine)
     sel = select(pg_attribute.c.attnum).where(
         and_(
             pg_attribute.c.attrelid == table_oid,
-            pg_attribute.c.attname == column_name
+            pg_attribute.c.attname.in_(column_names)
         )
     )
-    result = execute_statement(engine, sel, connection_to_use).fetchone()[0]
+    return execute_statement(engine, sel, connection_to_use).fetchall()
+
+
+def get_column_index_from_name(table_oid, column_name, engine, connection_to_use=None):
```

Comment:
I don't know if we want to do it on this PR, but we need to modify every place that the column index is used for identification and replace it with the `attnum`, or look up the name from `attnum` if needed.  The column index idea was a mistake, and is the source of numerous previous and a few visible current bugs.  For example, take a look at the `db.constraints.operations.select.get_column_constraints` function.  That will break if columns have been deleted and what not.

Reply:
I would prefer to do that in a different PR, once we get the interface right refactoring should be easier.

Reply:
Please open a separate issue for it if you're doing it in a different PR (just so we don't forget about it).